optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -585,6 +610,77 @@ Joining should only be used in cases where the other fact is actually required f
 the stream.
 ====
 
+[[constraintStreamsMappingTuples]]
+=== Mapping tuples
+
+Mapping enables you to transform each tuple in a constraint stream by applying a mapping function to it.
+The result of such mapping is `UniConstraintStream` of the mapped tuples.
+
+[source,java,options="nowrap"]
+----
+    private Constraint computerWithBiggestProcess(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(CloudProcess.class) // UniConstraintStream<CloudProcess>
+                .map(CloudProcess::getComputer)           // UniConstraintStream<CloudComputer>
+                ...
+    }
+----
+
+[NOTE]
+====
+In the example above, the mapping function produces duplicate tuples if two different `CloudProcess`es share a single `CloudComputer`.
+That is, such `CloudComputer` will appear in the resulting constraint stream twice.
+See <<constraintStreamsDistinct,`distinct()`>> for how to deal with duplicate tuples.
+====
+
+[[constraintStreamsMappingFunction]]
+==== Designing the mapping function
```

Comment:
(soft) the id differs from the title - so far the id, was usually <namespace of out section if needed> + <camelCase of title or at least the first few words of it>
I'd keep the title and the id very similar, so it's clear they are the same section.